nixos/tests/statsd.nix

```diff
@@ -33,8 +33,19 @@ with lib;
   };
 
   testScript = ''
-    $statsd1->start();
-    $statsd1->waitForUnit("statsd.service");
-    $statsd1->waitUntilSucceeds("nc -z 127.0.0.1 8126");
+    $machine->start();
+    $machine->waitForUnit("statsd.service");
+    $machine->waitForOpenPort(8126);
+
+    # check state of the `statsd` server
+    $machine->succeed('[ "health: up" = "$(echo health | nc 127.0.0.1 8126 -w 1)" ];');
```

Comment:
Also, I think this would be simpler, easier to read and more consistent with the rest:
```
$machine->succeed("echo health | nc -w 120 127.0.0.1 8126 -w 120 | grep -q 'health: up' ");
```